BUG: test_discrete_basic fails on scipy-1.13.1-cp311-cp311-win_arm64

### Describe your issue.

`scipy-1.13.1`, built from source against `numpy-2.0.0rc2` and `OpenBLAS-0.3.27` using `LLVM-18.1.6` on Python 3.11 for Windows on ARM64 (`win_arm64`), reproducibly fails the `test_discrete_basic[nbinom-arg14-False]` test (which is actually an amazing result given that this platform is not supported yet).

### Reproducing Code Example

```python
python -c "import scipy;scipy.test()"
```


### Error message

```shell
================================================= test session starts =================================================
platform win32 -- Python 3.11.9, pytest-8.2.1, pluggy-1.5.0
rootdir: C:\Users\gohlke
plugins: anyio-4.3.0, flaky-3.8.1, hypothesis-6.102.6, asyncio-0.23.7, cov-5.0.0, doctestplus-1.2.1, timeout-2.3.1
asyncio: mode=Mode.STRICT
collected 64315 items / 14574 deselected / 49741 selected

_lib\tests\test__gcutils.py ......                                                                               [  0%]
<snip>
stats\tests\test_variation.py ..............................                                                     [100%]

====================================================== FAILURES =======================================================
_______________________________________ test_discrete_basic[nbinom-arg14-False] _______________________________________
C:\Python311-arm64\Lib\site-packages\scipy\stats\tests\test_discrete_basic.py:47: in test_discrete_basic
    check_cdf_ppf(distfn, arg, supp, distname + ' cdf_ppf')
        arg        = (5, 0.5)
        distfn     = <scipy.stats._discrete_distns.nbinom_gen object at 0x0000018FE96DEAD0>
        distname   = 'nbinom'
        first_case = False
        m          = np.float64(5.0)
        rvs        = array([0, 2, 6, ..., 3, 3, 3])
        supp       = array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 21])
        v          = np.float64(10.0)
C:\Python311-arm64\Lib\site-packages\scipy\stats\tests\test_discrete_basic.py:198: in check_cdf_ppf
    npt.assert_array_equal(distfn.ppf(cdf_supp0, *arg),
        arg        = (5, 0.5)
        cdf_supp   = array([0.03125   , 0.109375  , 0.2265625 , 0.36328125, 0.5       ,
       0.62304688, 0.72558594, 0.80615234, 0.866577...7547913, 0.98455811, 0.99039459,
       0.99409103, 0.99640131, 0.99782825, 0.99870026, 0.99922806,
       0.99973324])
        cdf_supp0  = array([0.03125   , 0.109375  , 0.2265625 , 0.36328125, 0.5       ,
       0.62304687, 0.72558594, 0.80615234, 0.866577...7547913, 0.98455811, 0.99039459,
       0.99409103, 0.99640131, 0.99782825, 0.99870026, 0.99922806,
       0.99973324])
        distfn     = <scipy.stats._discrete_distns.nbinom_gen object at 0x0000018FE96DEAD0>
        msg        = 'nbinom cdf_ppf'
        supp       = array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 21])
C:\Python311-arm64\Lib\site-packages\numpy\_utils\__init__.py:85: in wrapper
    return fun(*args, **kwargs)
        args       = (array([ 0.,  1.,  2.,  3.,  4.,  5.,  7.,  7.,  8.,  9., 10., 11., 12.,
       13., 14., 15., 16., 17., 18., 19., 21....0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 21]), 'nbinom cdf_ppf-roundtrip')
        dep_version = '2.0.0'
        fun        = <function assert_array_equal at 0x0000018FE376F2E0>
        kwargs     = {}
        new_name   = 'desired'
        new_names  = ['actual', 'desired']
        old_name   = 'y'
        old_names  = ['x', 'y']
C:\Python311-arm64\Lib\contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError:
E   Arrays are not equal
E   nbinom cdf_ppf-roundtrip
E   Mismatched elements: 1 / 21 (4.76%)
E   Max absolute difference among violations: 1.
E   Max relative difference among violations: 0.16666667
E    ACTUAL: array([ 0.,  1.,  2.,  3.,  4.,  5.,  7.,  7.,  8.,  9., 10., 11., 12.,
E          13., 14., 15., 16., 17., 18., 19., 21.])
E    DESIRED: array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
E          17, 18, 19, 21])
        args       = (<built-in function eq>, array([ 0.,  1.,  2.,  3.,  4.,  5.,  7.,  7.,  8.,  9., 10., 11., 12.,
       13., 14., 15.,...., 18., 19., 21.]), array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 21]))
        func       = <function assert_array_compare at 0x0000018FE376F100>
        kwds       = {'err_msg': 'nbinom cdf_ppf-roundtrip', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x0000018FE1D57750>
================================================== warnings summary ===================================================
stats/tests/test_qmc.py::TestUtils::test_geometric_discrepancy_mst_with_zero_distances
  C:\Python311-arm64\Lib\site-packages\scipy\stats\tests\test_qmc.py:232: UserWarning: Sample contains duplicate points.
    assert_allclose(qmc.geometric_discrepancy(sample, method='mst'), 0.5)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================== short test summary info ===============================================
FAILED stats/tests/test_discrete_basic.py::test_discrete_basic[nbinom-arg14-False] - AssertionError:
=== 1 failed, 47175 passed, 2398 skipped, 14574 deselected, 155 xfailed, 12 xpassed, 1 warning in 575.70s (0:09:35) ===
Error in sys.excepthook:

Original exception was:
 WARNING: Maximum dimension of Krylov subspace exceeded prior to convergence. T
 ry increasing KMAX.
 neig =  0
```


### SciPy/NumPy/Python version and system information

```shell
Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 12:24:25) [MSC v.1938 64 bit (ARM64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); scipy.show_config()
1.13.1 2.0.0rc2 sys.version_info(major=3, minor=11, micro=9, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: internal
    found: true
    include directory: unknown
    lib directory: unknown
    name: dep2594698540304
    openblas configuration: unknown
    pc file directory: unknown
    version: 1.13.1
  lapack:
    detection method: internal
    found: true
    include directory: unknown
    lib directory: unknown
    name: dep2594698540304
    openblas configuration: unknown
    pc file directory: unknown
    version: 1.13.1
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: clang-cl
    linker: lld-link
    linker args: -Wl,-defaultlib:C:\Programs\LLVM18/lib/clang/18/lib/windows/clang_rt.builtins-aarch64.lib
    name: clang-cl
    version: 18.1.6
  c++:
    commands: clang-cl
    linker: lld-link
    linker args: -Wl,-defaultlib:C:\Programs\LLVM18/lib/clang/18/lib/windows/clang_rt.builtins-aarch64.lib
    name: clang-cl
    version: 18.1.6
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    args: --target=aarch64-pc-windows-msvc
    commands: flang
    linker: lld-link
    linker args: -Wl,-defaultlib:C:\Programs\LLVM18/lib/clang/18/lib/windows/clang_rt.builtins-aarch64.lib,
      --target=aarch64-pc-windows-msvc
    name: flang
    version: 17.0.6
  pythran:
    include directory: C:\Python311-arm64\Lib\site-packages/pythran
    version: 0.16.0
Machine Information:
  build:
    cpu: aarch64
    endian: little
    family: aarch64
    system: windows
  cross-compiled: false
  host:
    cpu: aarch64
    endian: little
    family: aarch64
    system: windows
Python Information:
  path: C:\Python311-arm64\python.exe
  version: '3.11'
```

--- a/scipy/stats/tests/test_discrete_basic.py
+++ b/scipy/stats/tests/test_discrete_basic.py
@@ -22,6 +22,8 @@ distdiscrete += [[stats.rv_discrete(values=vals), ()]]
 # For these distributions, test_discrete_basic only runs with test mode full
 distslow = {'zipfian', 'nhypergeom'}
 
+# Override number of ULPs adjustment for `check_cdf_ppf`
+roundtrip_cdf_ppf_exceptions = {'nbinom': 30}
 
 def cases_test_discrete_basic():
     seen = set()
@@ -193,8 +195,9 @@ def check_cdf_ppf(distfn, arg, supp, msg):
     cdf_supp = distfn.cdf(supp, *arg)
     # In very rare cases, the finite precision calculation of ppf(cdf(supp))
     # can produce an array in which an element is off by one.  We nudge the
-    # CDF values down by 15 ULPs help to avoid this.
-    cdf_supp0 = cdf_supp - 15*np.spacing(cdf_supp)
+    # CDF values down by a few ULPs help to avoid this.
+    n_ulps = roundtrip_cdf_ppf_exceptions.get(distfn.name, 15)
+    cdf_supp0 = cdf_supp - n_ulps*np.spacing(cdf_supp)
     npt.assert_array_equal(distfn.ppf(cdf_supp0, *arg),
                            supp, msg + '-roundtrip')
     # Repeat the same calculation, but with the CDF values decreased by 1e-8.

